File: cache/cache-annotation-processor/src/test/java/ru/tinkoff/kora/cache/annotation/processor/testdata/sync/CacheableSync.java
package ru.tinkoff.kora.cache.annotation.processor.testdata.sync;

import ru.tinkoff.kora.cache.annotation.CacheInvalidate;
import ru.tinkoff.kora.cache.annotation.CachePut;
import ru.tinkoff.kora.cache.annotation.Cacheable;
import ru.tinkoff.kora.cache.annotation.processor.testcache.DummyCache21;

import java.math.BigDecimal;
import java.util.Optional;

public class CacheableSync {

    public String value = "1";

    @Cacheable(DummyCache21.class)
    public String getValue(String arg1, BigDecimal arg2) {
        return value;
    }

    @Cacheable(DummyCache21.class)
    public Optional<String> getValueOptional(String arg1, BigDecimal arg2) {
        return value.describeConstable();
    }

    @CachePut(value = DummyCache21.class, parameters = {"arg1", "arg2"})
    public String putValue(BigDecimal arg2, String arg3, String arg1) {
        return value;
    }

    @CachePut(value = DummyCache21.class, parameters = {"arg1", "arg2"})
    public Optional<String> putValueOptional(BigDecimal arg2, String arg3, String arg1) {
        return Optional.ofNullable(value);
    }

    @CacheInvalidate(DummyCache21.class)
    public void evictValue(String arg1, BigDecimal arg2) {

    }

    @CacheInvalidate(value = DummyCache21.class, invalidateAll = true)
    public void evictAll() {

    }
}

File: cache/cache-symbol-processor/src/test/kotlin/ru/tinkoff/kora/cache/symbol/processor/testdata/CacheableSync.kt
package ru.tinkoff.kora.cache.symbol.processor.testdata

import ru.tinkoff.kora.cache.annotation.CacheInvalidate
import ru.tinkoff.kora.cache.annotation.CachePut
import ru.tinkoff.kora.cache.annotation.Cacheable
import ru.tinkoff.kora.cache.symbol.processor.testcache.DummyCache21
import java.math.BigDecimal

open class CacheableSync {
    var value = "1"

    @Cacheable(value = DummyCache21::class)
    open fun getValue(arg1: String?, arg2: BigDecimal?): String {
        return value
    }

    @CachePut(value = DummyCache21::class, parameters = ["arg1", "arg2"])
    open fun putValue(arg2: BigDecimal?, arg3: String?, arg1: String?): String {
        return value
    }

    @CacheInvalidate(value = DummyCache21::class)
    open fun evictValue(arg1: String?, arg2: BigDecimal?) {
    }

    @CacheInvalidate(value = DummyCache21::class, invalidateAll = true)
    open fun evictAll() {
    }

    // @Throws here is a typealias.
    // Should compile normally.
    // Method is not used in tests, only in processing
    @Throws
    @CacheInvalidate(value = DummyCache21::class, invalidateAll = true)
    open fun throws1() {
    }

    // @Throws here is a typealias.
    // Should compile normally.
    // Method is not used in tests, only in processing
    @Throws
    @CachePut(value = DummyCache21::class, parameters = ["arg1", "arg2"])
    open fun throws2(arg2: BigDecimal?, arg3: String?, arg1: String?): String {
        return value
    }

    // @Throws here is a typealias.
    // Should compile normally.
    // Method is not used in tests, only in processing
    @Throws
    @Cacheable(value = DummyCache21::class)
    open fun throws3(arg1: String?, arg2: BigDecimal?): String {
        return value
    }
}
